docs/wip/exchange-onboarding.md

```diff
@@ -0,0 +1,261 @@
+it standalone on any linux system built in the nix store or you can generate
+a docker container that can be ran on any docker container orchestrator like
+docker swarm or kubernetes.
+
+### Build and run in the nix store
+
+By default the wallet's local state goes in
+`./state-wallet-mainnet`.
+
+Build the wallet and generate the shell script to connect to
+mainnet (use `connectScripts.stagingWallet` for testnet)
+
+    nix-build -A connectScripts.mainnetWallet -o "./launch_$(date -I)_$(git rev-parse --short HEAD)"
+
+After the build finishes the generated connection script is
+available as a symlink called `./launch_2018-01-30_0d4f79eea`, or
+similar. Run that symlink as a script to start the wallet.
+
+### Build and run a docker image
+
+Follow the above instructions for customization and dependencies. To build a docker
+container and import the image run
+(use `connectScripts.stagingWallet` for testnet):
+
+    docker load < $(nix-build -A dockerImages.mainnetWallet)
+
+This will create an image `cardano-container-mainnet:latest`
+(or `cardano-container-staging:latest` for testnet)
+
+After this image is built, it can be used like any other docker image being pushed
+into a registry and pulled down using your preferred docker orchestration tool.
+
+The image can be ran using the following:
+
+    docker run --name cardano-mainnet-wallet --rm -it -p 127.0.0.1:8090:8090 -p 127.0.0.1:8000:8000 -v state-wallet-mainnet:/wallet cardano-container-mainnet:latest
+
+The above command will create a docker volume named `state-wallet-mainnet` and will mount
+that to /wallet. Note: if no volume is mounted to `/wallet` the container startup
+script will refuse to execute `cardano-node` and the container will exit.
+
+The location of `/wallet` cannot be changed, but you can mount any kind of volume
+you want in that directory that docker supports.
+
+Note that if you give this a different name than is specified above, use the
+name you used for any future docker commands in examples in the document.
+
+# Migrating from V0 to V1 API
+
+TODO: add instructions for migrating to new API
+
+# Usage FAQs
+
+## What are recommended hardware/software requirements for exchange wallets?
+
+RAM: 8 GB for building, 4 GB for running
+
+CPU: Modern x86_64 processor
+
+Disk: SSD recommended of 100 GB size with option to allocate more in future
+years.
+
+Operating System: NixOS or CentOS 7 recommended, although any linux
+distribution should work with nix package manager used for building
+the binaries
+
+Software Requirements: Nix package manager for building standalone
+and docker containers. On systems using docker, docker > 17.12 required.
+
+## How do I export the CA certificate for the API?
+
+The certificate is generated inside the wallet in the file `tls/server.cert`
+
+If you are using the docker container, this can be output using the command:
+
+`docker exec -it cardano-mainnet-wallet cat /wallet/state-wallet-mainnet/tls/server.cert`
+
```

Comment:
Just to clarify here, do we generate new certificates with each build? Meaning that exchanges A will end up with different and incompatible certificates than exchange B (not knowing anything about private keys of A)?

Reply:
We don't generate a CA at all right now, and a new cert/key are only regenerated if the tls directory doesn't exist in the wallet:
```
if [ ! -d ./cardano-state-staging/tls ]; then
  mkdir ./cardano-state-staging/tls/
  /nix/store/jn350na5wcshwsixpq1wbskdp7r3x0r3-openssl-1.0.2l-bin/bin/openssl req -x509 -newkey rsa:2048 -keyout ./cardano-state-staging/tls/server.key -out ./cardano-state-staging/tls/server.cert -days 3650 -nodes -subj "/CN=localhost"
fi
```